src/components/ClientEditDialog.js

```diff
@@ -78,6 +79,10 @@ const ClientEditDialog = (props) => {
         } else {
             setDisableEdit(false)
         }
+
+        if (client.currencyLocked) {
+            setClientCurrencyLock(true)
+        }
```

Comment:
Why do you need to add that attribute to a piece of state? I don't think that value is going to change over time with a user interaction so I would delete the `clientCurrecyLock and setClientCurrencyLock` and on the render function I would just type
`disabled={client.currencyLocked}`